Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -0,0 +1,357 @@
+  choose! U hxU hUl using fun x hx ↦ (nhds_basis_opens x).disjoint_iff_left.1 (H x hx)
+  choose hxU hUo using hxU
+  rcases hs.elim_nhds_subcover U fun x hx ↦ (hUo x hx).mem_nhds (hxU x hx) with ⟨t, htc, hts, hst⟩
+  refine (hasBasis_nhdsSet _).disjoint_iff_left.2
+    ⟨⋃ x ∈ t, U x, ⟨isOpen_biUnion fun x hx ↦ hUo x (hts x hx), hst⟩, ?_⟩
+  rw [compl_iUnion₂]
+  exact (countable_bInter_mem htc).mpr (fun i hi ↦ hUl _ (hts _ hi))
+
+/-- A filter `l` with the countable intersection property is disjoint with the neighborhood
+filter of a Lindelöf set if and only if it is disjoint with the neighborhood filter of each point
+of this set. -/
+theorem IsLindelof.disjoint_nhdsSet_right {l : Filter X} [CountableInterFilter l]
+    (hs : IsLindelof s) : Disjoint l (𝓝ˢ s) ↔ ∀ x ∈ s, Disjoint l (𝓝 x) := by
+  simpa only [disjoint_comm] using hs.disjoint_nhdsSet_left
+
+/-- For every family of closed sets whose intersection avoids a Lindelö set,
+there exists a countable subfamily whose intersection avoids this Lindelöf set. -/
+theorem IsLindelof.elim_countable_subfamily_closed {ι : Type v} (hs : IsLindelof s)
+    (t : ι → Set X) (htc : ∀ i, IsClosed (t i)) (hst : (s ∩ ⋂ i, t i) = ∅) :
+    ∃ u : Set ι, u.Countable ∧ (s ∩ ⋂ i ∈ u, t i) = ∅ := by
+    let U := tᶜ
+    have hUo : ∀ i, IsOpen (U i) := by simp only [Pi.compl_apply, isOpen_compl_iff]; exact htc
+    have hsU : s ⊆ ⋃ i, U i := by
+      simp only [Pi.compl_apply]
+      rw [← compl_iInter]
+      apply disjoint_compl_left_iff_subset.mp
+      simp only [compl_iInter, compl_iUnion, compl_compl]
+      apply Disjoint.symm
+      exact disjoint_iff_inter_eq_empty.mpr hst
+    rcases hs.elim_countable_subcover U hUo hsU with ⟨u, ⟨hucount, husub⟩⟩
+    use u, hucount
+    rw [← disjoint_compl_left_iff_subset] at husub
+    simp only [Pi.compl_apply, compl_iUnion, compl_compl] at husub
+    exact disjoint_iff_inter_eq_empty.mp (Disjoint.symm husub)
+
+/--To show that a Lindelöf set intersects the intersection of a family of closed sets,
+  it is sufficient to show that it intersects every countable subfamily. -/
+theorem IsLindelof.inter_iInter_nonempty {ι : Type v} (hs : IsLindelof s) (t : ι → Set X)
+    (htc : ∀ i, IsClosed (t i)) (hst : ∀ u : Set ι, u.Countable ∧ (s ∩ ⋂ i ∈ u, t i).Nonempty) :
+    (s ∩ ⋂ i, t i).Nonempty := by
+  contrapose! hst
+  rcases hs.elim_countable_subfamily_closed t htc hst with ⟨u, ⟨_, husub⟩⟩
+  use u
+  apply fun _ ↦ husub
+
+/-- For every open cover of a Lindelöf set, there exists a countable subcover. -/
+theorem IsLindelof.elim_countable_subcover_image {b : Set ι} {c : ι → Set X} (hs : IsLindelof s)
+    (hc₁ : ∀ i ∈ b, IsOpen (c i)) (hc₂ : s ⊆ ⋃ i ∈ b, c i) :
+    ∃ b', b' ⊆ b ∧ Set.Countable b' ∧ s ⊆ ⋃ i ∈ b', c i := by
+  simp only [Subtype.forall', biUnion_eq_iUnion] at hc₁ hc₂
+  rcases hs.elim_countable_subcover (fun i ↦ c i : b → Set X) hc₁ hc₂ with ⟨d, hd⟩
+  refine ⟨Subtype.val '' d, ?_, Countable.image hd.1 Subtype.val, ?_⟩-- d.image _, _⟩
+  · simp
```

Comment:
Another trick: you can inline the first subgoal via `by simp`. (Also, you can remove the commented code.)